server/src/main/java/org/opensearch/transport/nativeprotocol/NativeOutboundMessage.java

```diff
@@ -48,16 +55,17 @@
  *
  * @opensearch.internal
  */
-abstract class OutboundMessage extends NetworkMessage {
+public abstract class NativeOutboundMessage extends NetworkMessage implements ProtocolOutboundMessage {
 
     private final Writeable message;
 
-    OutboundMessage(ThreadContext threadContext, Version version, byte status, long requestId, Writeable message) {
+    NativeOutboundMessage(ThreadContext threadContext, Version version, byte status, long requestId, Writeable message) {
         super(threadContext, version, status, requestId);
         this.message = message;
     }
 
-    BytesReference serialize(BytesStreamOutput bytesStream) throws IOException {
+    @Override
+    public BytesReference serialize(BytesStreamOutput bytesStream) throws IOException {
```

Comment:
Please revert to package protected, it is not yet needed

Reply:
It is needed for tests like InboundPipelineTests etc.